owslib/catalogue/csw2.py

```diff
@@ -578,6 +580,12 @@ def _parserecords(self, outputschema, esn):
                 val = i.find(util.nspath_eval('gm03:fileIdentifier', namespaces))
                 identifier = self._setidentifierkey(util.testXMLValue(val))
                 self.records[identifier] = GM03(i)
+        elif MD_Metadata_3.handles(outputschema):  # ISO 19115 Part 3 XML
```

Comment:
It does check the root namespace, only I have put this function inside the class.
For ISO19115p3, in order to support both ESRI and geoserver, there are two versions of the set of namespaces.
I decided that the best place for this detection logic was inside the class.
I thought it would be more portable and maintainable to have it in there rather than outside the class.